### PR TITLE
Azure: use release-1.9 branch for CAPZ v1beta1

### PR DESCRIPTION
**Pull request by mboersma**
I updated all references to CAPZ's `release-1.8` branch to `release-1.9`, except when jobs weren't using kubekins `-1.27` or `-master`. So these should be compatible with Go 1.20.

### CODE REVIEW

**Comment by jackfrancis** [comment]
I audited these changes and didn't see any updated jobs that won't be able to accommodate CAPZ @ v1.9 (go 1.20 in particular).

/lgtm

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml:14]: this change seems like a mistake? it's in the file testing release-1.8

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.9.yaml
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.9.yaml:6]: I think the release-1.7 file needs to change to release-1.8 and release 1.8 to release-1.9 so we can keep the right intervals and tests for both (older release runs periodic less often)

**Comment by CecileRobertMichon** [comment]
> except when jobs weren't using kubekins -1.27 or -master

Can any of these jobs be updated to use a version of kubekins with go 1.20? If not, can we comment on those jobs so we know why they are using a different capz version in the future?

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml
Comment (jackfrancis) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml:14]: good catch

this is why my original attempt to do this kept a mutable "-minus-1" naming for these, handling all these changes with renaming the files and names and tabs every CAPZ release is going to be hard

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml:14]: would be great if we could generate these... just like we generate the k/k azure jobs

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.9.yaml
Comment (mboersma) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.9.yaml:6]: Oh, good call. I copied the 1.8 version and changed values to 1.9, but you're correct: will update.

**Comment by mboersma** [comment]
> Can any of these jobs be updated to use a version of kubekins with go 1.20? 

Probably, but I didn't want to expand the scope of this PR any further. I'll try to find out and add some comments or update them.

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml
Comment (mboersma) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml]: I fixed both mistakes by reverting to the original files, renaming the -1.8 job to -1.9, renaming the -1.7 job to -1.8, and doing the appropriate replacements within each.

Comment (jackfrancis) [reply config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics-v1beta1-release-1.8.yaml]: You make it sound easy!

**Comment by mboersma** [comment]
These are the jobs using `release-1.8` of CAPZ that this PR doesn't change:

- [ci-kubernetes-e2e-capz-master-containerd-windows-1-24](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml#L33)
- [pull-kubernetes-e2e-capz-windows-containerd-1-25](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.25-windows-presubmits.yaml#L23)
- [ci-kubernetes-e2e-capz-containerd-windows-1-25](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.25-windows.yaml#L33)
- [ci-kubernetes-e2e-capz-containerd-windows-serial-slow-1-25](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.25-windows.yaml#L75)
- [pull-kubernetes-e2e-capz-windows-containerd-1-26](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.26-windows-presubmits.yaml#L22)
- [ci-kubernetes-e2e-capz-master-containerd-windows-1-26](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.26-windows.yaml#L23)

@jsturtevant @marosset each of these jobs is using a specific tagged version of the `kubekins` image. I assume they shouldn't be updated to `release-1.9` and Go 1.20, but let me if I'm wrong. I might add a `# Don't update to a newer branch` comment to make it clearer for the next person to update this.

**Comment by mboersma** [comment]
@CecileRobertMichon @jackfrancis I *think* we want to leave the jobs I listed above alone, so this PR should be complete and ready for another review.

File: config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: does k/k have a plan to backport go 1.20 to previous releases as they started doing for the past go version? if so this is only true until kubekins-e2e images get 1.20

Comment (mboersma) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: No idea. I looked around in test-infra but didn't find any clues, but I can try to find out.

Should I close this PR and open separate ones per-project? I'm concerned that trying to update projects other than cluster-api-provider-azure is delaying testing with the new release branch.

Comment (jackfrancis) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: My read on that backporting was that it was a one-time thing, and not an ongoing thing.

Comment (jackfrancis) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: I think the only sustainable way forward is to not use kubekins for these CAPZ jobs

Comment (mboersma) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: What if I just changed the comment to:

`# Don't update from release-1.8 without Go 1.20 support in the container image`

Comment (CecileRobertMichon) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: It was actually an enhancement: https://github.com/kubernetes/enhancements/issues/3744

And looks like backports are coming soon for go-1.20: https://github.com/kubernetes/release/issues/2815#issuecomment-1526400628

In this case @mboersma let's merge this for now and bump the jobs once 1.20 is part of the older kubekins images. Would you be able to please clarify the comments that bumping is TODO dependent on go 1.20 being backported to the test images?

Comment (jackfrancis) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: Thanks for tracking that down @CecileRobertMichon, happy to learn k/k is going to solve this for us this time :)

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by mboersma** [comment]
This needs a rebase now, because some underlying jobs were changed yesterday by new PRs. (Several others were created and merged without conflicts.) Their authors were assigned to review this PR.

I think I'll close this PR and open separate ones per-project. That will allow CAPZ to move forward, and other projects can choose to update their test-infra jobs if and when they are interested.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

File: config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml
Comment (marosset) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: 1.24 needs to stay on `release-1.6`.
We have problems building the Windows container images in the cloud-provider-azure release-1.24 branch

https://github.com/kubernetes/test-infra/pull/29468

Comment (mboersma) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: It looks like it's already been updated to `release-1.8` in #29248.

Should I remove this `TODO` comment (or all of them)? Or change it do:
`# Don't update to release-1.9`

Comment (marosset) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: I updated it yesterday to fix the periodic jobs.

test-infra @ master currently has

```yaml
  extra_refs:
  - org: kubernetes-sigs
    repo: cluster-api-provider-azure
    # The Windows 1.24 jobs should stay on CAPZ release-1.6 for now because later versions build CNM and CCM
    # images from k-sigs/cluster-provider-azure repo and the container image builds are not working.
    base_ref: release-1.6
    path_alias: sigs.k8s.io/cluster-api-provider-azure
    workdir: true
```

Comment (marosset) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml:33]: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-1.24-windows.yaml#L72-L79

**Comment by mboersma** [comment]
Rebased again to incorporate #29489.

**Comment by mboersma** [comment]
@andyzhangx @feiskyer could you review this when you have time? It needs approval from remaining project owners.

**Comment by lzhecheng** [comment]
/approve
for cloud-provider-azure

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by mboersma** [comment]
Thanks everyone for helping me polish this up and get it merged! I'll keep an eye on testgrid, but please let me know if you see any new issues testing with CAPZ release-1.9.

**Pull request merged**
